App/Http/Controllers/TaskController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Repositories\Eloquent\TaskRepository;
+use App\Repositories\Eloquent\UserRepository;
+use App\Models\Task;
+use Illuminate\Support\Facades\Auth;
+
+
+class TaskController extends Controller
+{
+  public function __construct()
+  {
+      $this->middleware('auth');
+  }
+
+
+  public function index()
+  {
+      $taskRepository = new TaskRepository();
+      $user = Auth::user();
+      $tasks= $taskRepository->findAllBy('owner_user_id',$user->id);
+
+      return view('tasks.index',compact("tasks"));
+  }
+
+  public function onAssign($id)
+  {
+       $taskRepository = new TaskRepository();
+       $task= $taskRepository->find($id);
```

Comment:
failsafe checks and redirect user back to the task list saying this task doesn't exist

Reply:
Apply this on all places

Reply:
done